src/content/en/2022/javascript.md

```diff
@@ -16,7 +16,864 @@ featured_stat_2: TODO
+
+#### Bytes
+
+So we know that hosts are shipping a lot of third-party scripts, but what's the byte cost of first- versus third-party scripts?
+
+{{ figure_markup(
+    image="bytes-party.png",
+    caption="Distribution of first- versus third-party JavaScript bytes by host.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=1411726363&format=interactive",
+    sheets_gid="1560368150",
+    sql_file="bytes_by_3p.sql"
+  )
+}}
+
+At nearly every percentile, the amount of bytes third-party scripts ship exceeds that of first-party scripts. At the 75th percentile, it appears that third-party script payloads are _twice_ that of first-party scripts. At the 90th percentile, it appears that the amount of third-party scripts sent over the wire is nearly _one megabyte_.
+
+If you find your website's first versus third-party script payloads is similar to the graph above, it is key that you should work with your engineering organization to try and get this number down. It can only help your users if you do.
+
+### Dynamic `import()`
+
+[Dynamic `import()`](https://v8.dev/features/dynamic-import) is a variant of the [static `import` syntax](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/import) that can be run anywhere in a script, whereas static `import` expressions must be run at the top of a JavaScript file and nowhere else.
+
+Dynamic `import()` allows developers to effectively "split" off chunks of JavaScript code from their main bundles to be loaded on-demand, which can improve startup performance by loading less JavaScript upfront.
+
+{{ figure_markup(
+    caption="The percentage of mobile pages using dynamic `import()`.",
+    content="0.34%",
+    classes="big-number",
+    sheets_gid="563835654",
+    sql_file="dynamic_import.sql"
+  )
+}}
+
+A staggeringly low 0.34% of all observed mobile pages currently use dynamic `import()`, while 0.41% of desktop pages use it. This is a missed opportunity to ship less code during startup. Perhaps dynamic `import()` is not seeing much use because it shifts the loading of JavaScript on-demand for features, rather than upfront.
+
+It's tricky, but a balance _can_ be struck, and it involves gauging the user's intent. One way of deferring loading of JavaScript without delaying interactions is to [preload](https://developer.mozilla.org/docs/Web/HTML/Link_types/preload) that JavaScript when the user signals intent to make an interaction. One example of this could be to defer loading JavaScript for the validation of a form, and preload that JavaScript once the user has focused a field in that form. That way, when the JavaScript is requested, it will already be in the browser cache.
```

Comment:
```suggestion
It's tricky, but a balance can be struck, and it involves gauging the user's intent. One way of deferring loading of JavaScript without delaying interactions is to [preload](https://developer.mozilla.org/docs/Web/HTML/Link_types/preload) that JavaScript when the user signals intent to make an interaction. One example of this could be to defer loading JavaScript for the validation of a form, and preload that JavaScript once the user has focused a field in that form. That way, when the JavaScript is requested, it will already be in the browser cache.
```